go/inst/write_buffer.go

```diff
@@ -0,0 +1,150 @@
+	P75InstancesWritten               float64
+	P95InstancesWritten               float64
+	P99InstancesWritten               float64
+	MaxWriteLatencySeconds            float64 // metrics for time to write each set of instances
+	MeanWriteLatencySeconds           float64
+	MedianWriteLatencySeconds         float64
+	P75WriteLatencySeconds            float64
+	P95WriteLatencySeconds            float64
+	P99WriteLatencySeconds            float64
+}
+
+// WriteBufferAggregatedSince returns the aggregated for metrics in the collection since the specified time.
+func WriteBufferAggregatedSince(c *collection.Collection, t time.Time) WriteBufferAggregate {
+	var (
+		writeTimings []float64
+		writeItems   []float64
+	)
+
+	// Retrieve values since the time specified
+	values, err := c.Since(t)
+	wba := WriteBufferAggregate{
+		InstanceFlushIntervalMilliseconds: config.Config.InstanceFlushIntervalMilliseconds,
+		InstanceWriteBufferSize:           config.Config.InstanceWriteBufferSize,
+	}
+
+	if err != nil {
+		return wba // empty data
+	}
+
+	// generate the metrics
+	for _, v := range values {
+		writeTimings = append(writeTimings, v.(*WriteBufferMetric).WriteLatency.Seconds())
+		writeItems = append(writeItems, float64(v.(*WriteBufferMetric).Items))
+		wba.SumInstancesWritten += v.(*WriteBufferMetric).Items
+		if v.(*WriteBufferMetric).Err != nil {
+			wba.ErrorCount++
+			wba.SumInstancesErrors += v.(*WriteBufferMetric).Items
+		}
+	}
+
+	wba.WriteCount = len(writeTimings)
+
+	// generate aggregate timing metrics
+	if s, err := stats.Max(stats.Float64Data(writeTimings)); err == nil {
+		wba.MaxWriteLatencySeconds = s
+	}
+	if s, err := stats.Mean(stats.Float64Data(writeTimings)); err == nil {
+		wba.MeanWriteLatencySeconds = s
+	}
+	if s, err := stats.Median(stats.Float64Data(writeTimings)); err == nil {
+		wba.MedianWriteLatencySeconds = s
+	}
+	if s, err := stats.Percentile(stats.Float64Data(writeTimings), 75); err == nil {
+		wba.P75WriteLatencySeconds = s
+	}
+	if s, err := stats.Percentile(stats.Float64Data(writeTimings), 95); err == nil {
+		wba.P95WriteLatencySeconds = s
+	}
+	if s, err := stats.Percentile(stats.Float64Data(writeTimings), 99); err == nil {
+		wba.P99WriteLatencySeconds = s
+	}
+	// generate aggregate item size metrics
+	if s, err := stats.Max(stats.Float64Data(writeItems)); err == nil {
+		wba.MaxInstancesWritten = s
+	}
+	if s, err := stats.Mean(stats.Float64Data(writeItems)); err == nil {
+		wba.MeanInstancesWritten = s
+	}
+	if s, err := stats.Median(stats.Float64Data(writeItems)); err == nil {
+		wba.MedianInstancesWritten = s
+	}
+	if s, err := stats.Percentile(stats.Float64Data(writeItems), 75); err == nil {
+		wba.P75InstancesWritten = s
+	}
+	if s, err := stats.Percentile(stats.Float64Data(writeItems), 95); err == nil {
+		wba.P95InstancesWritten = s
+	}
+	if s, err := stats.Percentile(stats.Float64Data(writeItems), 99); err == nil {
+		wba.P99InstancesWritten = s
+	}
+
+	return wba
+}
```

Comment:
It doesn't work on non-integer values. I guess I could ensure I'm using the time.Duration values and work on that and then convert any final results into the ..Seconds values I'm using but that's not how I implemented this.

If you'd accept the PR as is I'll make an issue to look to remove the use of this extra package if that can be done.